pkg/build/controller/controller.go

```diff
@@ -317,7 +317,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			}
 		}
 	case kapi.PodFailed:
-		build.Status.Reason = ""
+		// NOTE: why do we need to reset the reason for a failed Pod???
+		// build.Status.Reason = ""
```

Comment:
so you're saying the flow is:

within the pod, it realizes something has failed, updates the build status to failed w/ reason
then the pod gets marked failed
then the controller sees the pod is failed and updates the build reason to "", wiping out your reason

?